cs-rft/src/main/java/io/cloudslang/content/rft/actions/ftp/Get.java

```diff
@@ -0,0 +1,92 @@
+import static com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType.ERROR;
+import static com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType.RESOLVED;
+import static io.cloudslang.content.constants.OutputNames.*;
+import static io.cloudslang.content.constants.ResponseNames.FAILURE;
+import static io.cloudslang.content.constants.ResponseNames.SUCCESS;
+import static io.cloudslang.content.rft.utils.ftp.Constants.*;
+import static io.cloudslang.content.rft.utils.ftp.Descriptions.Common.*;
+import static io.cloudslang.content.rft.utils.ftp.FTPInputsValidation.verifyInputsFTP;
+import static io.cloudslang.content.rft.utils.ftp.Inputs.*;
+import static io.cloudslang.content.utils.OutputUtilities.getFailureResultsMap;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+public class Get {
+    @Action(name = "FTP Get Operation",
+            outputs = {
+                    @Output(value = RETURN_RESULT, description = RETURN_RESULT_DESC),
+                    @Output(value = RETURN_CODE, description = RETURN_CODE_DESC),
+                    @Output(value = EXCEPTION, description = EXCEPTION_DESC),
+                    @Output(value = FTP_REPLY_CODE, description = FTP_REPLY_CODE_DESC),
+                    @Output(value = FTP_SESSION_LOG, description = FTP_SESSION_LOG_DESC)
+            },
+            responses = {
+                    @Response(text = SUCCESS, field = RETURN_CODE, value = ReturnCodes.SUCCESS, matchType = COMPARE_EQUAL, responseType = RESOLVED, description = SUCCESS_DESC),
+                    @Response(text = FAILURE, field = RETURN_CODE, value = ReturnCodes.FAILURE, matchType = COMPARE_EQUAL, responseType = ERROR, description = FAILURE_DESC)
+            })
+    public Map<String, String> execute(@Param(value = PARAM_HOSTNAME, description = PARAM_HOSTNAME_DESC) String hostName,
+                                       @Param(value = PARAM_PORT, description = PARAM_PORT_DESC) String port,
+                                       @Param(value = PARAM_LOCAL_FILE, description = PARAM_LOCAL_FILE_DESC) String localFile,
+                                       @Param(value = PARAM_REMOTE_FILE, description = PARAM_REMOTE_FILE_DESC) String remoteFile,
+                                       @Param(value = PARAM_USER, description = PARAM_USER_DESC) String user,
+                                       @Param(value = PARAM_PASSWORD, description = PARAM_PASSWORD_DESC) String password,
+                                       @Param(value = PARAM_TYPE, description = PARAM_TYPE_DESC) String type,
+                                       @Param(value = PARAM_PASSIVE, description = PARAM_PASSIVE_DESC) String passive,
+                                       @Param(value = PARAM_CHARACTER_SET, description = PARAM_CHARACTER_SET_DESC) String characterSet) {
+
+        hostName = defaultIfEmpty(hostName, EMPTY);
```

Comment:
For required inputs it's no need to assing them empty and verify after that if they are empty.

Reply:
Fixed